xof/blake2xs/blake.go

```diff
@@ -61,6 +65,13 @@ func (x *xof) Reseed() {
 	x.impl = y.(*xof).impl
 }
 
+func (x *xof) Reset() {
+	x.impl.Reset()
+	if len(x.seed) > blake2s.Size {
+		x.impl.Write(x.seed[blake2s.Size:])
+	}
+}
```

Comment:
Is it really necessary to store the entire seed then? Wouldn't storing just `seed2` be enough and test here if its len is bigger than 0?

Reply:
Now that I am not using New() in Reset() as you suggested, you are right we can do that. Fixed !